src/paymaster.ts

```diff
@@ -0,0 +1,37 @@
+import { getAddress } from "ethers";
+import { ApprovalBasedPaymasterInput, GeneralPaymasterInput } from "./types";
+import { encodeData, getPaymasterParams } from "./utils";
+
+export class Paymaster {
+    type: string;
+    address: string;
+    paymasterInput: ApprovalBasedPaymasterInput | GeneralPaymasterInput;
+
+    constructor(
+        type: "ApprovalBased" | "General",
+        address: string,
+        tokenAddress: string = "",
+        minimalAllowance: number = 1,
+        innerInput: any[] = [],
+    ) {
+        this.type = type;
+        this.address = getAddress(address);
+        const encodedInnerInput = encodeData(innerInput);
+        this.paymasterInput =
+            type == "ApprovalBased"
+                ? {
+                      type,
+                      token: getAddress(tokenAddress),
+                      minimalAllowance,
+                      innerInput: encodedInnerInput,
+                  }
+                : {
+                      type,
+                      innerInput: encodedInnerInput,
+                  };
+    }
+
+    getPaymasterParams() {
```

Comment:
Why do we need this class if we simply want to build paymaster params? Shouldn't it be a util instead?

Reply:
Actually I also notice this method will require adding parameters as those are required for the `ApprovalBased` paymasters that accept a quantity of tokens to cover the transaction cost. 

As for making this a class, this class will also include a method to enrich a transaction with the data for `paymasterParams` which is useful when developers only want to sponsor some transactions. Something like `.addParamsToTransaction(trx)` that calls `getPaymasterParams` to fill in the transaction object. This removes the need for the developer to manipulate the transaction object manually. 

Not precious about this either though, so if you think strongly of keeping these as functions then we could go that way but I find this: 

```ts
let tx = contract.methodWrite(...);
tx = utils.addPaymasterParamsToTransaction(utils.getPaymasterParams(...));
```

Less appealing than:

```ts
const paymaster = new Paymaster(...);
let tx = contract.methodWrite(...);
tx = paymaster.addParamsToTransaction(tx);
```